src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IProjectContext.cs

```diff
@@ -7,26 +7,34 @@
 namespace Microsoft.VisualStudio.LanguageServices.ProjectSystem
 {
     /// <summary>
-    /// Project context to initialize properties and items of a Workspace project created with <see cref="IProjectContextFactory.CreateProjectContext(string, string, string, Guid, string, Shell.Interop.IVsHierarchy, CommandLineArguments)"/>. 
+    /// Project context to initialize properties and items of a Workspace project created with <see cref="IProjectContextFactory.CreateProjectContext(string, string, string, Shell.Interop.IVsHierarchy)"/>. 
     /// </summary>
-    internal interface IProjectContext
+    internal interface IProjectContext : IDisposable
     {
         // Project properties.
-        ProjectId Id { get; }
         string DisplayName { get; set; }
         string ProjectFilePath { get; set; }
         Guid Guid { get; set; }
         string ProjectType { get; set; }
+        bool DesignTimeBuildStatus { get; set; }
 
         // Options.
         void SetCommandLineArguments(CommandLineArguments commandLineArguments);
+        void SetCommandLineArguments(string commandLineForOptions);
```

Comment:
I am probably confused about this: _We need to decide what happens when I give you options you don't understand._

Do you think we will have project options that cannot be understood by the default language parser, but can still be represented by CommandLineArguments OR we will have project options that are not even representable in CommandLineArguments?
If you meant the former, then CPS will need to use the API that takes CommandLineArguments. Otherwise, only the string based API makes sense, and we need to just do some additional parsing in AbstractProject to set these non-compilation options.